c_binding/utils.c

```diff
@@ -114,3 +114,19 @@ int _read_file(const char *path, uint8_t *buff, ssize_t *size, size_t max_size)
     }
     return rc;
 }
+
+/* Find the first trailing space character,
+ * insert a NULL character in its place.
+ */
+void _trim_trailing_spaces(char *string_ptr)
+{
+    int i = 0;
+
+    while (string_ptr[i] != '\0') {
+        if (string_ptr[i] == 0x20) {
+            string_ptr[i] = '\0';
+            break;
+        }
+        ++i;
+    }
```

Comment:
The SPC 3,4,5 indicates:

```
The PRODUCT SERIAL NUMBER field contains right-aligned ASCII data (see 4.3.1) that is a vendor 
specific serial number.
```

So You should remove both leading space[1] and trailing space. And meanwhile, the SPC allows space as serial numbe: 
```
ASCII data fields shall contain only ASCII printable characters (i.e., code values 20h 
through 7Eh) and may be terminated with one or more ASCII null (00h) characters.
```

[1]: The LSI expander use leading space padding:
```
[fge@megadeth ~]$ sudo sg_vpd --page=0x80 /dev/sde -H
Unit serial number VPD page:
 00     00 80 00 14 20 20 20 20  20 57 44 2d 57 43 43 34    ....     WD-WCC4
 10     4a 45 44 41 58 45 53 5a                             JEDAXESZ
[fge@megadeth ~]$ sudo sg_vpd --page=0x80 /dev/sde 
Unit serial number VPD page:
  Unit serial number:      WD-WCC4JEDAXESZ
```